### PR TITLE
Fix: `indent` shouldn't check the last line unless it is a punctuator

### LINKED ISSUE #3498: Indent issue with multi-line var/const
I'm having an issue with indentation, before I was using `0.24` and it was all good but now with multi-line var declarations I get indent errors. I may perhapse be using the wrong config but from the docs this is what I understood to be the config that would allow for declarations like I have.

### Config
`"indent": [2, 2, {"VariableDeclarator": 2}],`

#### Tried
`"indent":  [2, 2, {"VariableDeclarator": { "var": 2, "let": 2, "const": 3}}]`

### Source file 
```javascript
export default function() {
  const HO = 'bah',
        TE = 'mah'

  var res,
      a = 5,
      b = 4

  if (a++ === b) {
    let multiplier = HO + TE

    multiplier += '329' 

    res = multiplier
  }

  return res
}

```

### Error
`9:3  error  Expected indentation of 6 space characters but found 2  indent`

### PR DESCRIPTION
**Pull request by BYK**
Fixes #3498

### CODE REVIEW

**Comment by gyandeeps** [comment]
The test added right now is not sufficient enough for the issue we are trying to solve. We need something after the `var` to prove that. Thats why I would suggest to use the example from the issue but you can remove the export part rather just take the content from the function.

**Comment by BYK** [comment]
@gyandeeps - it actually is enough because of the trailing new line at the end. I first write a failing test, and then fix it :)

I can add the exact case from the ticket but I'm not sure if it's really worth it.

**Comment by BYK** [comment]
@gyandeeps - Welp, I added one more case anyways :)

**Comment by gyandeeps** [comment]
Why I asked is because it was giving errors on the line after the `var` statement.

**Comment by BYK** [comment]
>Why I asked is because it was giving errors on the line after the var statement.

Yeah and I'm saying the test case had one line after the `var` statement so it was still giving an error saying I needed to indent the blank line (or something like that).

**Comment by gyandeeps** [comment]
Got it. Didnt know about that.
Thanks

**Comment by BYK** [comment]
Meh, np. Mostly my bad since it goes against the expected test case pattern. Hence the extra test case. Thanks for bringing it up :)

File: tests/lib/rules/indent.js [hunk 1]
```diff
@@ -718,6 +718,30 @@ ruleTester.run("indent", rule, {
                 "}\n",
             ecmaFeatures: { modules: true, defaultParams: true },
             options: [2]
+        },
+        {
+            code:
+                "const HO = 'bah',\n" +
```
Comment (nzakas) [inline tests/lib/rules/indent.js]: Slight nitpick: The word "ho" has a negative connotation in the United States. Can you change it to something else?

Comment (BYK) [inline tests/lib/rules/indent.js]: Done.

**Comment by gyandeeps** [comment]
LGTM

**Pull request merged**